### PR TITLE
extended test cases, pom updates, misc. refactorings

### PR DESCRIPTION
**Pull request by lazerion**

### CODE REVIEW

File: src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java [hunk 1]
```diff
@@ -51,45 +55,94 @@
...
+    private static final int VERIFICATION_WAIT_TIMEOUT = 5;
+    private static final int DISCOVERY_RETRY_TIMEOUT = 1;
 
     private final EurekaClient eurekaClient;
-    private final DiscoveryManager discoveryManager;
     private final DynamicPropertyFactory dynamicPropertyFactory;
     private final ApplicationInfoManager applicationInfoManager;
 
     private final boolean selfRegistration;
-    private final boolean clientMode;
```
Comment (noctarius) [inline src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java]: Why remove the final?

File: src/main/java/com/hazelcast/eurekast/one/EurekastOneProperties.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
  * <p>Configuration class of the Hazelcast Discovery Plugin for <a href="https://github.com/Netflix/eureka">Eureka 1</a>.</p>
  * <p>For possible configuration properties please refer to the public constants of this class.</p>
  */
-public final class EurekastOneProperties {
+final class EurekastOneProperties {
```
Comment (noctarius) [inline src/main/java/com/hazelcast/eurekast/one/EurekastOneProperties.java]: The idea is to have those classes public to generate Javadoc for it. Javadoc can act as the documentation of available properties in this case. It was on purpose :)

File: src/test/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategyTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        verify(applicationInfoManager).setInstanceStatus(InstanceInfo.InstanceStatus.UP);
+    }
+
+    @Test
+    public void shouldNotRegisterWhenClientModeEnabled(){
+        when(eurekaClient.getApplication(APPLICATION_NAME))
+                .thenReturn(new Application());
+        reset(applicationInfoManager);
+
+        strategy.setClientMode(true);
```
Comment (noctarius) [inline src/test/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategyTest.java]: I think this should be a separate test class, to remove the final from clientMode just to put it into the same testcase is not too nice :)

File: src/main/java/com/hazelcast/eurekast/one/EurekastOneProperties.java (hunk 2 above)
Comment (lazerion) [inline src/main/java/com/hazelcast/eurekast/one/EurekastOneProperties.java]: thanks on it

File: src/test/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategyTest.java (hunk 3 above)
Comment (lazerion) [inline src/test/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategyTest.java]: thanks on it

**Comment by lazerion** [comment]
@noctarius I updated the PR could you please review and merge if it checks out?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+- Add *eureka-client.properties* file to working directory or use `eureka.client.props` dynamic property to define 
+property file path without `properties` extension.
+
+The following is an example declarative configuration.
+
+```xml
+ <hazelcast>
+   ...
+    <properties>
+        <property name="hazelcast.discovery.enabled">true</property>
```
Comment (noctarius) [inline README.md]: I think we can remove it from the docs now, as it was a temp property only (and as far as I know discovery is now enabled by default - we need to verify though :))

File: checkstyle/suppressions.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    <!-- Suppress strict duplicate code checking -->
+    <suppress checks="StrictDuplicateCode" files="\.java" lines="1-15"/>
+
+    <!-- Exclude implementation packages from JavaDoc checks -->
+    <suppress checks="JavadocPackage" files="/impl/"/>
+    <suppress checks="JavadocType" files="/impl/"/>
+    <suppress checks="JavadocVariable" files="/impl/"/>
+
+    <!-- AWS -->
+    <suppress checks="JavadocType" files="com/hazelcast/aws/"/>
```
Comment (noctarius) [inline checkstyle/suppressions.xml]: aws?

File: src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java [hunk 6]
```diff
@@ -51,45 +55,94 @@
 class EurekastOneDiscoveryStrategy
```
Comment (noctarius) [inline src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java]: Yeah, i started it as a private project and eurekast was a nice mixture of eureka and hazelcast. We can go ahead and name it Hazelcast Eureka1 Discovery, just to align it with the other ones.

File: src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java [hunk 7]
```diff
@@ -16,80 +16,125 @@
...
+    private final String namespace;
+
+    @VisibleForTesting
+    EurekastOneDiscoveryStrategy(EurekaClient eurekaClient,
+                                 ApplicationInfoManager manager,
+                                 boolean clientMode) {
+        super(new NoLogFactory().getLogger(EurekastOneDiscoveryStrategy.class.getName()),
+                Collections.<String, Comparable>emptyMap());
+
+        this.selfRegistration = getOrDefault(EUREKAST_ONE_SYSTEM_PREFIX, SELF_REGISTRATION, true);
```
Comment (noctarius) [inline src/main/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategy.java]: If we rename, don't forget the properties :)

File: src/main/resources/apache-v2-license.txt [hunk 8]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
```
Comment (noctarius) [inline src/main/resources/apache-v2-license.txt]: As far as I know, common practice is LICENSE in root

**Comment by noctarius** [comment]
We can actually merge it first and apply the little changes afterwards, just as you prefer :)

File: checkstyle/suppressions.xml (hunk 5 above)
Comment (lazerion) [inline checkstyle/suppressions.xml]: oops, fixed, thanks

File: README.md (hunk 4 above)
Comment (lazerion) [inline README.md]: ITs are failing w/out it.

**Comment by lazerion** [comment]
@noctarius I believe, I addressed all possible review feedbacks. can you merge?

**Pull request merged**